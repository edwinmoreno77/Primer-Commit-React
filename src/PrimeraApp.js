
// import { Fragment } from "react";
import PropTypes from 'prop-types';
// import React, { Fragment } from "react";// v17

//FC
const PrimeraApp = ({ saludo, subtitulo }) => {

    // const saludo = 'Hola Mundo';
    // console.log(props)
    // {
    //     nombre: 'Edwin',
    //     apellido: 'Moreno'// ejm: JSON.stringify
    // }

    return (
        <>
            <h1>{saludo}</h1>
            {/*<pre>{JSON.stringify(saludo, null, 3)}</pre>*/}
            <p>{subtitulo}</p>
        </>
    );
}



//     return (
//         <Fragment>
//             <h1>hola mundo</h1>
//             <p>Mi Primera App</p>
//         </Fragment>
//     );
// } Este es un ejm de forma corta de fragment.



PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}
export default PrimeraApp;