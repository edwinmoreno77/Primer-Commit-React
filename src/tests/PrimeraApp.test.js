import React from 'react';
import '@testing-library/jest-dom';
import { render, shallow } from 'enzyme';
// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { getByAltText, getByText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';


describe('Pruebas en PrimeraApp', () => {

    // test('Debe de mostrat el mensaje "Hola, soy Goku"', () => {

    //     const saludo = 'Hola, soy Goku';

    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('<PrimeraApp/> Debe hacer match con snapshot', () => {

        const saludo = 'hola, soy goku';
        // const wrapper = shallow(<PrimeraApp saludo={saludo} />);//otra forma de hacerlo

        // expect(wrapper).toMatchSnapshot();
        const { container } = render(<PrimeraApp saludo={saludo} />);

        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Goku';
        const subTitulo = 'Soy un subtitulo';

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subTitulo} />);

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subTitulo);
    });


})