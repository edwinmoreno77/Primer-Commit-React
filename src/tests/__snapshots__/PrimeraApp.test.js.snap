// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp <PrimeraApp/> Debe hacer match con snapshot 1`] = `undefined`;
